src/main/java/org/wildfly/security/auth/server/SecurityDomain.java

```diff
@@ -289,6 +289,12 @@ public SecurityIdentity authenticate(Principal principal, Evidence evidence) thr
         if (principal != null) serverAuthenticationContext.setAuthenticationPrincipal(principal);
         if (serverAuthenticationContext.verifyEvidence(evidence)) {
             if (serverAuthenticationContext.authorize()) {
+                Credential credential = evidence.asCredential();
```

Comment:
OK, going to change this. I have considered checking for specific types and only support the types we define there. But I thought that this could bring some more flexibility and give Evidence providers the means to automatically propagate their corresponding credentials.

I also need to consider BearerTokenEvidence. Noticed that it is missing ...